### PR TITLE
Fix filter errors

### PR DESCRIPTION
**Pull request by zackgalbreath**
Fix various issues I noticed while playing around with filters & subprojects.

### CODE REVIEW

File: api/v1/index.php [hunk 1]
```diff
@@ -1160,21 +1176,25 @@ function get_child_builds_hyperlink($parentid, $filterdata)
...
-        $filter['field'] != 'stamp' &&
-        $filter['compare'] != 0 &&
-        $filter['compare'] != 20 &&
-        $filter['compare'] != 40 &&
-        $filter['compare'] != 60 &&
-        $filter['compare'] != 80) {
+      $filter['field'] != 'site' &&
+      $filter['field'] != 'stamp' &&
+      $filter['compare'] != 0 &&
+      $filter['compare'] != 20 &&
```
Comment (jamiesnape) [inline api/v1/index.php]: No need to do this now, but really need to make these named constants.

File: filterdataFunctions.php [hunk 2]
```diff
@@ -931,7 +930,7 @@ function get_filterdata_from_request($page_id = '')
 
         $sql_field = $pageSpecificFilters->getSqlField($field);
 
-        if ($fieldtype == 'date') {
+        if ($field == 'buildstarttime') {
```
Comment (jamiesnape) [inline filterdataFunctions.php]: `===` would be preferable, but `==` is fine given the code elsewhere.

**Comment by jamiesnape** [comment]
LGTM.

**Comment by zackgalbreath** [comment]
Thanks for the feedback, @jamiesnape .  Agreed on both accounts.

**Pull request merged**